ddht/cli_commands.py

```diff
@@ -23,3 +24,15 @@ async def do_main(boot_info: BootInfo) -> None:
     logger.info("Started main process (pid=%d)", os.getpid())
     async with background_trio_service(app) as manager:
         await manager.wait_finished()
+
+
+async def do_crawl(boot_info: BootInfo) -> None:
+
+    if boot_info.protocol_version is not ProtocolVersion.v5:
+        raise Exception("Currently crawling is only supported on the v5 network.")
```

Comment:
I think you *could* raise `argparse.ArgumentError` here.

Reply:
Turned it into a `ValueError`. I tried turning it into an `ArgumentError` but didn't have easy access to the parser its constructor wants.